### PR TITLE
Enable Setting Button Text Color

### PR DESCRIPTION
**Pull request by dri94**
A dialog's text button color should not always be derived from the styles.xml. In branding cases dialog's may have a more custom look throughout the system. Therefore allowing a custom attribute to be set to dialog's will enable better theming.

### CODE REVIEW

File: core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt [hunk 1]
```diff
@@ -196,7 +196,8 @@ class MaterialDialog(
         textRes = res,
         text = text,
         fallback = android.R.string.ok,
-        typeface = this.buttonFont
+        typeface = this.buttonFont,
+            textColor = R.attr.md_button_text_color
```
Comment (afollestad) [inline core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt]: Too much indent here

File: core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt [hunk 2]
```diff
@@ -236,7 +237,8 @@ class MaterialDialog(
         textRes = res,
         text = text,
         fallback = android.R.string.cancel,
-        typeface = this.buttonFont
+        typeface = this.buttonFont,
+            textColor = R.attr.md_button_text_color
```
Comment (afollestad) [inline core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt]: Same here

File: core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt (hunk 1 above)
Comment (petedoyle) [reply core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt]: Brand new 2.0 user, may be off-

In order to match `md_color_title` and `md_color_content`, would it make sense to call this `md_color_button`? 

Or, possibly:
`md_color_button_positive` and `md_color_button_negative`.

File: core/src/main/res-public/values/public.xml [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <attr format="color" name="md_color_title"/>
   <attr format="color" name="md_color_content"/>
   <attr format="color" name="md_ripple_color"/>
+  <attr format="color" name="md_color_button_text"/>
```
Comment (afollestad) [inline core/src/main/res-public/values/public.xml]: Looks like the value here doesn't match what you put in styles.xml

File: core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt [hunk 4]
```diff
@@ -236,7 +237,8 @@ class MaterialDialog(
         textRes = res,
         text = text,
         fallback = android.R.string.cancel,
-        typeface = this.buttonFont
+        typeface = this.buttonFont,
+        textColor = R.attr.md_button_text_color
```
Comment (afollestad) [inline core/src/main/java/com/afollestad/materialdialogs/MaterialDialog.kt]: This also doesn't match the name in public.xml

**Pull request merged**